02_bubble/jimbotech/bubble_test.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+)
+
+func TestNotModifiedInput(t *testing.T) {
+	input := []int{64, 11, 1, 1, 2, 3, 5, 8, 13, 21, 34}
+	original := make([]int, len(input))
+	copy(original, input)
+
+	bubble(input)
+
+	if !reflect.DeepEqual(original, input) {
+		t.Errorf("The input slice was altered %v does not match %v", original, input)
+	}
+}
+func TestBubble(t *testing.T) {
+	input := []int{64, 11, 1, 1, 2, 3, 5, 8, 13, 21, 34}
+	expected := []int{1, 1, 2, 3, 5, 8, 11, 13, 21, 34, 64}
+
+	sortedResult := bubble(input)
+
+	if !reflect.DeepEqual(sortedResult, expected) {
+		t.Errorf("Returned value %v does not match %v", sortedResult, expected)
+	}
+}
+func TestNoInput(t *testing.T) {
+	input := []int{}
```

Comment:
The nil slice and the empty slice are effectively the same thing in Go. Underlying memory allocation may differ but the behaviour is the same.

However, a test for a 1-element slice is definitely a good idea since the code being tested does an `i+1` in the loop, you want to make sure you have your boundary conditions correct.

Reply:
Done